### DISCUSSION TITLE
custom nginx timeout

### DISCUSSION THREAD

**Discussion by waterrowernohrd** [beyondcode/herd-community#261] [Q&A]
Has someone managed to increase the nginx timeout? For me it is and will be 60 seconds.

**Comment by vahidirp** [comment]
Yes, the timeout settings in Nginx can be adjusted to increase the maximum allowed time for certain operations. The relevant directive for controlling timeouts in Nginx is typically timeout or proxy_timeout, depending on the context.

If you are dealing with proxying requests to another server (e.g., with proxy_pass), you might want to adjust the proxy_timeout directive. For example:
```
location / {
    proxy_pass http://backend_server;
    proxy_timeout 120s;  
}
```
This would set the timeout for proxy connections to 120 seconds. You can adjust the timeout value according to your needs.

If you're dealing with other types of timeouts, like the client timeout or the server timeout, you might need to adjust different directives. For instance:
```
server {
    listen 80;
    client_body_timeout 120s;  # Set the client body timeout to 120 seconds
    client_header_timeout 120s;  # Set the client header timeout to 120 seconds
    send_timeout 120s;  # Set the send timeout to 120 seconds
}
```
